### PR TITLE
chore!: Address strict concurrency warnings

### PR DESCRIPTION
**Pull request by sichanyoo**
## Issue \#

https://github.com/awslabs/aws-sdk-swift/issues/1414

## Description of changes

Address strict concurrency warnings thrown by building with 
```
swift build --build-tests -Xswiftc -strict-concurrency=complete
```

This PR addresses warnings that will become compile errors in Swift 6.

Changes fall under one of the following:
- Adding conformance to Sendable protocol
- Adding @preconcurrency tag to C library import
- Changing global variable into a static function that returns that value

## Scope

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Sources/ClientRuntime/Telemetry/Logging/SDKLoggingSystem.swift [hunk 1]
```diff
@@ -8,18 +8,18 @@
...
         let label = logHandlerFactory.label
         factories[label] = logHandlerFactory
     }
 
-    public static func initialize(defaultLogLevel: SDKLogLevel = .info) async {
+    public func initialize(defaultLogLevel: SDKLogLevel = .info) async {
         if isInitialized { return } else { isInitialized = true }
         LoggingSystem.bootstrap { label in
-            if let factory = factories[label] {
+            if let factory = self.factories[label] {
```
Comment (jbelkins) [inline Sources/ClientRuntime/Telemetry/Logging/SDKLoggingSystem.swift:22]: This is still throwing an actor-isolation warning (see below)

**Review by jbelkins — CHANGES_REQUESTED** [review]
See one remaining Swift 6 actor-isolation warning

File: Sources/ClientRuntime/Telemetry/Logging/SDKLoggingSystem.swift (hunk 1 above)
Comment (jbelkins) [reply Sources/ClientRuntime/Telemetry/Logging/SDKLoggingSystem.swift:22]: (view the file to see the warning)

Comment (sichanyoo) [reply Sources/ClientRuntime/Telemetry/Logging/SDKLoggingSystem.swift:22]: Huh this is weird, when building locally I don't get this error. Regardless, will address it.

**Review by jbelkins — APPROVED** [review]

**Pull request merged**